src/mesos_state_client.erl

```diff
@@ -61,20 +67,34 @@ try_token_from_file() ->
       try_read_token_file(Path)
   end.
 
--spec(try_read_token_file(string()) -> {ok, string()} | undefined).
+-spec(try_read_token_file(string()) -> {ok, string()} | {error, no_token_parsed} | undefined).
 try_read_token_file(Path) ->
   case file:read_file(Path) of
     {ok, Binary} ->
-      {ok, strip_token_binary(Binary)};
+      parse_token(Binary);
     {error, Error} ->
       lager:warning("unable to open configured token file ~p: ~p", [Path, Error]),
       undefined
   end.
 
--spec(strip_token_binary(binary()) -> string()).
-strip_token_binary(Binary) ->
-  Stripped = binary:replace(Binary, ?WHITESPACE, <<>>, [global]),
-  binary_to_list(Stripped).
+-spec(parse_token(binary()) -> {ok, string()} | {error, no_token_parsed}).
+parse_token(<<>>) ->
+  Path = application:get_env(?APP, ssl_token_file, test_path),
+  lager:error("failed to parse auth token from configured file: ~p", [Path]),
+  {error, no_token_parsed};
+parse_token(B) ->
```

Comment:
this is written for parsing multi-line env var files for if that becomes reality